katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        DataSet.__init__(self, filename, ref_ant, time_offset)
+
+        # Load file
+        self.file, self.version = H5DataV2_5._open(filename, mode)
+        f = self.file
+
+        # Load main HDF5 groups
+        data_group    = f["Data"]
+        config_group  = f["MetaData/Configuration"]
+        sensors_group = f["MetaData/Sensors"]
+        markup_group  = f["Markup"]
+
+        # Get observation script parameters, with defaults
+        for k, v in config_group['Observation'].attrs.iteritems():
+            self.obs_params[str(k)] = v
+        self.observer      = self.obs_params.get('observer', '')
+        self.description   = self.obs_params.get('description', '')
+        self.experiment_id = self.obs_params.get('experiment_id', '')
+
+        # ------ Extract timestamps ------
+        accumulation_length = get_single_value(config_group['DBE'], 'accum_length')  # Accumulation length in number of FPGA frames.
+        coarse_size = get_single_value(config_group["DBE"], "dbe.fft.coarse.size")
+        fine_size = get_single_value(config_group["DBE"], "dbe.fft.fine.size")
+        sampling_frequency = 800e6  # Have to hardcode this for the time being.
+        self.dump_period  = float(accumulation_length) * coarse_size / sampling_frequency
+        if fine_size != 0:
+            self.dump_period *= fine_size
+
+        # Obtain visibility data and timestamps
+        self._vis         = data_group['VisData'][:-1,:,:]
+        self._stokes      = data_group['StokesData']
+        self._timestamps  = data_group['Timestamps']
+        self._time_av_ll  = data_group["Left Power time average"]
+        self._time_av_rr  = data_group["Right Power time average"]
+        self._time_av_q   = data_group["Stokes Q time average"]
+        self._time_av_u   = data_group["Stokes U time average"]
+        num_dumps         = len(self._timestamps)
+        if num_dumps != self._vis.shape[0]:
+            raise BrokenFile('Number of timestamps received '
+                        '(%d) differs from number of dumps in data (%d)' % (num_dumps, self._vis.shape[0]))
+
+        # Do quick test for uniform spacing of timestamps (necessary but not sufficient)
+        expected_dumps = (self._timestamps[num_dumps - 1] - self._timestamps[0]) / self.dump_period + 1
+        # The expected_dumps should always be an integer (like num_dumps), unless the timestamps and/or dump period
+        # are messed up in the file, so the threshold of this test is a bit arbitrary (e.g. could use > 0.5)
+        irregular = abs(expected_dumps - num_dumps) >= 0.01
+        if irregular:
+            # Warn the user, as this is anomalous
+            logger.warning(("Irregular timestamps detected in file '%s': "
+                           "expected %.3f dumps based on dump period and start/end times, got %d instead") %
+                           (filename, expected_dumps, num_dumps))
+            if quicklook:
```

Comment:
Do you still need quicklook for AVN? In other words, are your timestamps interspersed throughout the HDF5 file or written as a single dataset at the end?

Reply:
I remember you talked about this. It's written as a single datafile at the end. Should this just be removed?

Reply:
Our original HDF5 files took a long while to open with katdal (if very large), mostly because the timestamps have to be read and the timestamps were interspersed throughout the file. In v3 we wrote the timestamps as a single contiguous dataset just before closing the file, which allowed katdal to open the file quickly.

So if your files already open "quickly" (say within 1-2 seconds) with `katdal.open`, I would remove the whole quicklook machinery.

Reply:
Removed in d8ea341.